js/blockedLinks.js

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/* globals gPinnedLinks, gUserDatabase, async*/
+"use strict";
+
+/**
+ * Singleton that keeps track of all blocked links in the grid.
+ */
+(function(exports) {
+  const gBlockedLinks = {
+    /**
+     * The cached list of blocked links.
+     */
+    _links: {},
+
+
+    /**
+     * Load the blocked links from gUserDatabase and cache them.
+     */
+    initBlockedLinks() {
+      var loadPromise = gUserDatabase.load("prefs", "blockedLinks");
+      loadPromise.then(loadedLinks => {
+        this._links = (loadedLinks && loadedLinks.length) ? JSON.parse(loadedLinks) : {};
+      });
+      return loadPromise;
+    },
+
+    /**
+     * The list of blocked links.
+     */
+    get links() {
+      return this._links;
+    },
+
+    _hash(url) {
+      var buffer = new TextEncoder("utf-8").encode(url);
+      return window.crypto.subtle.digest("SHA-256", buffer).then(digest => {
```

Comment:
No particular reason except that it may be more difficult to crack. I suppose our point of hashing is to obfuscate the URLs so SHA-1 is sufficient